designs/disruption-controls-by-reason.md

```diff
@@ -0,0 +1,312 @@
+2. AllowedDisruptionByReason = minNodeCountOfActiveBudget[reason] - unhealthyNodes - totalDisruptingNodes
+The second equation is the one we should opt into.
+
+Take this budget as an example
+```yaml
+budgets: 
+  - nodes: 15 
+    reasons: [drifted, underutilized]
+    schedule: "* * * * *"
+  - nodes: 10
+    reasons: [drifted]
+    schedule: "* * * * *"
+  - nodes: 5 
+    schedule: "* * * * *" 
+```
+First, calculate minNodeCountOfActiveBudget. Given there are two budgets for drifted, we select the lower number:
+
+```
+minNodeCountOfActiveBudget = {
+  drifted: 10 
+  underutilized: 15 
+  Default: 5
+}
+```
+
+In the first equation, tracking the number of nodes disrupted by each reason is required. For example:
+```
+disrupting = {
+  drifted: 3 
+  underutilized: 6
+  Expriation: 3
+  empty: 2
+}
+```
+
+Assuming zero unhealthy nodes:
+**First Equation**
+- drifted = 10 - 3: Disruption allowed as drifted bucket isn't full.
+- underutilized = 15 - 6: Disruption allowed since 15 - 6 > 0.
+- Default = 5 - expired - empty = 0: No disruption allowed for reasons other than consolidation and drift.
+
+**Second Equation**
+- Total Disrupting = 3 + 6 + 5 = 14.
+- drifted = 10 - 14: No disruption allowed due to drift.
+- underutilized = 15 - 14: Disruption allowed for at least one node due to consolidation.
+- Default: 5 - 14: No disruption allowed for other methods.
+
+#### Decision
+The second equation simplifies reasoning for cluster operators. A node can be tagged for multiple disruption reasons (e.g., empty, drifted, underutilized), making it challenging to precisely calculate in-flight disruptions. The second equation streamlines this process by providing a clearer view of the overall disruption impact.
+### Q: How should we handle an unspecified default reason? 
+```yaml
+budgets: 
+  - nodes: 10
+    reasons: [drift, underutilized]
+    schedule: "* * * * *"
+  - nodes: 5 
+    reasons: [empty] 
+    schedule: "* * * * *" 
+```
+In the case of a budget like above, default is undefined. Should karpenter assume the user doesn't want to disrupt any other reasons? Or should we assume that if a default is unspecified, they want us to disrupt anyway?  
+The intuitive options if there is no active default budget is to allow disruption of either 0 or total number of nodes(meaning unbounded disruption).
+Lets choose total number of nodes, since this allows the user to also specify periods where no nodes are to be disrupted of a particular type of disruption, and makes more sense with the existing karpenter behavior today.
+
+# API Design
+## Approach A: Extending the Budget API to specify a reason 
+This section contrasts two approaches for specifying disruption reasons in the v1beta1 nodepool API: a list of reasons (List Approach) and a single reason per budget (Single Reason Approach).
+
+### List Approach: Multiple Reasons per Budget
```

Comment:
Worth putting something like this to guide the users that this is what you're thinking of.
```suggestion
### List Approach: Multiple Reasons per Budget - Recommended
```